### ISSUE TITLE
Add support for reporting location to server URL when specific GCM/APN arrives

### ISSUE THREAD

**Issue by uriva** [phonegap/phonegap-plugin-push#6]
Hi,
I have code that adds the following functionality (as of now just on android):
Upon a GCM with REPORT_LOCATION in a certain field, the client sends its current location via http request to a predetermined endpoint (without bothering the user).
(It can't be done from javascript because the app is not always active)
This is useful when the server needs accurate client location in a specific time, but periodically updating the server costs precious battery.

Does this sound like something you'd like to incorporate?
Is this plugin currently up and and working?

**Comment by macdonst** [comment]
@uriva Yes, this plugin is currently up and working. It sounds interesting but I'd like to see support for Android, iOS and soon to be Windows Phone before adding it as a new feature.

**Comment by uriva** [comment]
gotcha

**Comment by macdonst** [comment]
@uriva you don't need to close this issue but we could move it to an enhancement.

**Comment by uriva** [comment]
Reopening as I see my current master push-plugin repository is deprecated, so I prefer to add to this one.
I can write only the Android case, but I don't have knowledge / tools for iOS or windows.
Is anybody up for it?

**Comment by fredgalvao** [comment]
The cold-start background notification feature, specifically the event proposed [here](https://github.com/phonegap/phonegap-plugin-push/issues/93#issuecomment-141408654) would already make that possible without having to resort to extra native code, and would already work on all platforms. And I'd rather it be on javascript side, so that any customizations to the request made to the server could be done easily.

**Comment by uriva** [comment]
iiuc, this functionality is currently available in android, and the issue is just for ios?

**Comment by fredgalvao** [comment]
No, unfortunately it's not available in any platform yet.

**Comment by uriva** [comment]
/:

**Comment by uriva** [comment]
There are two issues that prevents #93 from being a complete solution:

1. The functionality isn't available if a user "ends task" through the square button (no deal breaker)
2. If you call `navigator.geolocation.getCurrentPosition` from the pushPlugin callback, it will not work until the user has entered the app.

I know that you guys are aware of the first issue, but perhaps the second one is new information. If you have any ideas how to bypass this that would be great, e.g. if we can make android think this is foreground javascript somehow (I assume that natively that call would have been allowed from a background process?)
